airflow/contrib/hooks/sagemaker_hook.py

```diff
@@ -219,6 +219,52 @@ def create_tuning_job(self, tuning_job_config):
         return self.conn.create_hyper_parameter_tuning_job(
             **tuning_job_config)
 
+    def create_transform_job(self, transform_job_config, wait_for_completion=True):
+        """
+        Create a transform job
+        :param transform_job_config: the config for transform job
+        :type transform_job_config: dict
+        :param wait_for_completion:
+        if the program should keep running until job finishes
+        :type wait_for_completion: bool
+        :return: A dict that contains ARN of the transform job.
+        """
+        if self.use_db_config:
+            if not self.sagemaker_conn_id:
+                raise AirflowException(
+                    "SageMaker connection id must be present to \
+                    read SageMaker transform job configuration.")
+
+            sagemaker_conn = self.get_connection(self.sagemaker_conn_id)
+
+            config = sagemaker_conn.extra_dejson.copy()
+            transform_job_config.update(config)
+
+        self.check_for_url(transform_job_config
+                           ['TransformInput']['DataSource']
+                           ['S3DataSource']['S3Uri'])
+
+        response = self.conn.create_transform_job(
+            **transform_job_config)
+        if wait_for_completion:
+            self.check_status(['InProgress', 'Stopping', 'Stopped'],
```

Comment:
```

    def check_status(self, non_terminal_states,
                     failed_state, key,
                     describe_function, *args):
        """
        :param non_terminal_states: the set of non_terminal states
        :type non_terminal_states: dict
        :param failed_state: the set of failed states
        :type failed_state: dict
        :param key: the key of the response dict
        that points to the state
        :type key: string
        :param describe_function: the function used to retrieve the status
        :type describe_function: python callable
        :param args: the arguments for the function
        :return: None
        """
```

The `non_terminal_states` and `failed_state` should be `dict`'s according to the docs. Can we make these one `set`'s as well? `{'InProgress', 'Stopping', 'Stopped'}`